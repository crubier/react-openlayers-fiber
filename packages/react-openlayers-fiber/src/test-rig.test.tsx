import { render } from "@testing-library/react";
import { Map } from "ol";
import React from "react";

describe("Test setup", () => {
  it("can run basic test", () => {
    expect(1).toBe(1);
  });
  it("can run inline snapshot test", () => {
    expect(1).toMatchInlineSnapshot(`1`);
  });
  it("can run dom snapshot test", () => {
    const { container } = render(<div>ok</div>);
    expect(container).toMatchSnapshot();
  });
  it("can run react testing library test", () => {
    const { getByText } = render(<div>ok</div>);
    const heading = getByText(/ok/i);
    console.log(heading);
    expect(heading).toMatchInlineSnapshot(`
<div>
  ok
</div>
`);
  });
  it("can run ol tests", () => {
    const map = new Map({});
    expect(map.getAllLayers().length).toBe(0);
  });
});
