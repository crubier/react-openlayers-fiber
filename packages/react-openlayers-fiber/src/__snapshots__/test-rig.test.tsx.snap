// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test setup > can run dom snapshot test 1`] = `
<div>
  <div>
    ok
  </div>
</div>
`;

exports[`Test setup can run dom snapshot test 1`] = `
<div>
  <div>
    ok
  </div>
</div>
`;
